service/src/main/java/org/jboss/sbomer/service/feature/sbom/errata/event/CommentAdvisoryOnRelevantEventsListener.java

```diff
@@ -0,0 +1,360 @@
+import org.jboss.sbomer.core.features.sbom.utils.ObjectMapperProvider;
+import org.jboss.sbomer.core.features.sbom.utils.SbomUtils;
+import org.jboss.sbomer.service.feature.FeatureFlags;
+import org.jboss.sbomer.service.feature.sbom.errata.ErrataClient;
+import org.jboss.sbomer.service.feature.sbom.errata.dto.Errata;
+import org.jboss.sbomer.service.feature.sbom.errata.dto.enums.ErrataStatus;
+import org.jboss.sbomer.service.feature.sbom.k8s.model.SbomGenerationStatus;
+import org.jboss.sbomer.service.feature.sbom.model.Sbom;
+import org.jboss.sbomer.service.feature.sbom.model.SbomGenerationRequest;
+import org.jboss.sbomer.service.feature.sbom.service.SbomService;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.ObservesAsync;
+import jakarta.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+
+@ApplicationScoped
+@Slf4j
+public class CommentAdvisoryOnRelevantEventsListener {
+
+    @ConfigProperty(name = "SBOMER_ROUTE_HOST", defaultValue = "sbomer")
+    String sbomerHost;
+
+    @Inject
+    @RestClient
+    ErrataClient errataClient;
+
+    @Inject
+    SbomService sbomService;
+
+    @Inject
+    FeatureFlags featureFlags;
+
+    public void onRequestEventStatusUpdate(@ObservesAsync RequestEventStatusUpdateEvent event) {
+        log.debug("Event received for request event status update...");
+        if (!featureFlags.errataCommentsGenerationsEnabled()) {
+            log.warn(
+                    "Errata comments generation feature is disabled, no comments will be added to the Errata advisory!!");
+            return;
+        }
+
+        if (!isValidRequestEvent(event)) {
+            return;
+        }
+
+        ErrataAdvisoryRequestConfig config = (ErrataAdvisoryRequestConfig) event.getRequestEventConfig();
+        Errata errata = errataClient.getErratum(config.getAdvisoryId());
+        if (!isValidErrata(errata, config.getAdvisoryId())) {
+            return;
+        }
+
+        if (event.getRequestEventId() != null) {
+            // Advisories which produced request events which handled the manifestation.
+            // Standard advisories and text-only advisories with "deliverables" notes content fall in this category.
+            handleAutomatedManifestationAdvisory(event, config);
+        } else {
+            // Advisories whose manifestation was handled autonomously.
+            // Text-only advisories with "manifest" notes content fall in this category.
+            handleManualManifestationAdvisory(errata, config);
+        }
+    }
+
+    private void handleAutomatedManifestationAdvisory(
+            RequestEventStatusUpdateEvent event,
+            ErrataAdvisoryRequestConfig config) {
+        List<V1Beta1RequestRecord> requestRecords = sbomService
+                .searchAggregatedResultsNatively("id=" + event.getRequestEventId());
```

Comment:
Do we specify explicit order for this query? For example last are first in the result list.

Reply:
Yes the sorting is done here: https://github.com/project-ncl/sbomer/blob/a97133c8f3eb4624d285a3ebce1a4228d57f012d/service/src/main/java/org/jboss/sbomer/service/feature/sbom/service/RequestEventRepository.java#L436

So it's for receivalTime in descending order (we will get last requests first)